parquet/src/main/java/org/apache/iceberg/parquet/ReadConf.java

```diff
@@ -59,42 +61,46 @@
   private final boolean reuseContainers;
   @Nullable
   private final Integer batchSize;
+  @Nullable
+  private final NameMapping nameMapping;
 
   // List of column chunk metadata for each row group
   private final List<Map<ColumnPath, ColumnChunkMetaData>> columnChunkMetaDataForRowGroups;
 
   @SuppressWarnings("unchecked")
   ReadConf(InputFile file, ParquetReadOptions options, Schema expectedSchema, Expression filter,
            Function<MessageType, ParquetValueReader<?>> readerFunc, Function<MessageType,
-           VectorizedReader<?>> batchedReaderFunc, boolean reuseContainers,
+           VectorizedReader<?>> batchedReaderFunc, NameMapping nameMapping, boolean reuseContainers,
            boolean caseSensitive, Integer bSize) {
     this.file = file;
     this.options = options;
     this.reader = newReader(file, options);
     MessageType fileSchema = reader.getFileMetaData().getSchema();
-
     boolean hasIds = ParquetSchemaUtil.hasIds(fileSchema);
-    MessageType typeWithIds = hasIds ? fileSchema : ParquetSchemaUtil.addFallbackIds(fileSchema);
+    this.nameMapping = nameMapping == null ? MappingUtil.create(expectedSchema) : nameMapping;
```

Comment:
This is not the right place to infer a name mapping. This class should apply the name mapping if it exists, and use a position-based fallback otherwise.

Reply:
Agreed.